pkg/infrastructure/clusterapi/clusterapi.go

```diff
@@ -0,0 +1,275 @@
+package clusterapi
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	utilkubeconfig "sigs.k8s.io/cluster-api/util/kubeconfig"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager/signals"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/cluster/metadata"
+	"github.com/openshift/installer/pkg/asset/ignition/bootstrap"
+	"github.com/openshift/installer/pkg/asset/ignition/machine"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/asset/kubeconfig"
+	"github.com/openshift/installer/pkg/asset/manifests/capiutils"
+	capimanifests "github.com/openshift/installer/pkg/asset/manifests/clusterapi"
+	"github.com/openshift/installer/pkg/clusterapi"
+	"github.com/openshift/installer/pkg/infrastructure"
+	"github.com/openshift/installer/pkg/types"
+)
+
+// InfraProvider is the base implementation for provisioning cluster
+// infrastructure using CAPI. Platforms should embed this struct and
+// implement:
+// .
+type InfraProvider struct {
+	CAPIInfraHelper
+}
+
+func InitializeProvider(platform CAPIInfraHelper) infrastructure.Provider {
+	return InfraProvider{platform}
+}
+
+type CAPIInfraHelper interface {
```

Comment:
```suggestion
type Provider interface {
```

or similar?

Reply:
+1 to improving the name. Any other suggestions from other reviewers?

Reply:
My suggestion is `CAPILifecycleEvents`. Any `helper` related name risks to be too generic